tests/roles/backend_services/tasks/main.yaml

```diff
@@ -87,6 +87,21 @@
       args:
         chdir: "../config"
 
+    - name: template out the OpenStackVersion deployment with container overrides
+      ansible.builtin.template:
+        src: openstack_version.j2
+        dest: "../config/openstack_version_overrides.yaml"
+        mode: '644'
+        force: true
+
+    - name: Apply OpenStackVersion with container overrides to environment
```

Comment:
thank you! this was also suggested in slack chat.

i am getting this however, i tried both as you see above and then moved the task to before the create controlplane task and in both cases have this:

```
TASK [backend_services : Apply OpenStackVersion with container overrides to environment] ***
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "set -euxo pipefail\n\n\noc apply -f openstack_version_overrides.yaml\n", "delta": "0:00:00.623117", "end": "2024-04-19 10:31:00.738986", "msg": "non-zero return code", "rc": 1, "start": "2024-04-19 10:31:00.115869", "stderr": "+ oc apply -f openstack_version_overrides.yaml\nerror: error when retrieving current configuration of:\nResource: \"core.openstack.org/v1beta1, Resource=openstackversions\", GroupVersionKind: \"core.openstack.org/v1beta1, Kind=OpenStackVersion\"\nName: \"\", Namespace: \"openstack\"\nfrom server for: \"openstack_version_overrides.yaml\": resource name may not be empty", "stderr_lines": ["+ oc apply -f openstack_version_overrides.yaml", "error: error when retrieving current configuration of:", "Resource: \"core.openstack.org/v1beta1, Resource=openstackversions\", GroupVersionKind: \"core.openstack.org/v1beta1, Kind=OpenStackVersion\"", "Name: \"\", Namespace: \"openstack\"", "from server for: \"openstack_version_overrides.yaml\": resource name may not be empty"], "stdout": "", "stdout_lines": []}
 
```


logs there

https://logserver.rdoproject.org/95/52995/2/check/periodic-data-plane-adoption-osp-17-to-extracted-crc/afee6c3/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/logs/test_with_ceph_out_2024-04-19T10:00:31EDT.log

Reply:
I think this might be related to the ansible template that is applied.
It might just need `metadata.name` and `metadata.namespace`. We can try on a live environment to double check.

[1] https://logserver.rdoproject.org/95/52995/2/check/periodic-data-plane-adoption-osp-17-to-extracted-crc/afee6c3/controller/ci-framework-data/logs/openstack-k8s-operators-openstack-must-gather/namespaces/openstack/crs/openstackversions.core.openstack.org/openstack.yaml

[2] https://logserver.rdoproject.org/95/52995/2/check/periodic-data-plane-adoption-osp-17-to-extracted-crc/afee6c3/controller/data-plane-adoption-tests-repo/data-plane-adoption/tests/config/openstack_version_overrides.yaml

Reply:
I'm also not sure we need the targetVersion to be `0.0.1` like the default CR: we might want to clone the existing data from the CR shipped by default.

Reply:
ack i had the same thought about removing the target version altogether if that is OK to do at the time we are creating this, will try in the next update

Reply:
thanks added into metadata and now OK